### ISSUE TITLE
`object_property` uses modified names for boolean getters

### ISSUE THREAD

**Issue by Sbte** [ronaldoussoren/pyobjc#586]
**Describe the bug**

The getter of a `_C_BOOL` `object_property` is prefixed with `is` [here](https://github.com/ronaldoussoren/pyobjc/blob/43a4f9dbe3b7879146650a812629934332187f98/pyobjc-core/Lib/objc/_properties.py#L200) but the setter is not. This means that when defining the `condition` `object_property` in my objective-c header as 
```
@property BOOL isCondition;
```
the `setIsCondition:` selector will not be found when trying to set the property value.

This can be fixed by using
```
@property (setter=setCondition:) BOOL isCondition;
```
but in this case KVO is broken because KVO gets generated for `condition`, not `isCondition`.

What does work (for getting, setting and KVO) is implementing an `isCondition` `object_property` and then using
```
@property (getter=isIsCondition) BOOL isCondition;
```
But now I have this oddly named `isIsCondition` getter.

So in short, I'm not sure why the getter is prefixed with `is`, but it's causing issues with setters and KVO.

Edit: This also seems to break the `depends_on` parameter.

**Comment by ronaldoussoren** [comment]
The API is based on this pattern in system headers on macOS:

```objective-c
@property (readonly, getter=isRemote) BOOL remote
```

That is, the name of the property does not have the "is" prefix, only the getter method does.

**Comment by Sbte** [comment]
I fixed my issues by not using the `@getter` decorator, but instead defining the property after the getter and adding the getter to the property by calling the [getter](https://github.com/ronaldoussoren/pyobjc/blob/43a4f9dbe3b7879146650a812629934332187f98/pyobjc-core/Lib/objc/_properties.py#L291) method manually. So the issues weren't caused by the modified names as I initially thought.

The actual issue with the `@getter` decorator seems to be that the property is defined twice if the name of the getter does not match the name of the property, and that that breaks the class definition.

Since this is probably a very niche use case and I'm not even completely sure what the actual issue is, I'm not sure if it's worth it to open a new issue for this.

Anyway, thanks for the fast initial reply.
